From 1a74985b2a404639b08882c57f3147229605dfd5 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 19 Apr 2016 06:50:31 -0400
Subject: [PATCH] extlinux: pull in sys/sysmacros.h for major/minor/makedev

These functions are defined in sys/sysmacros.h, so add the include to
main.c.  This is already handled correctly in mountinfo.c.  Otherwise
we get build failures like:

main.o: In function 'find_device_sysfs':
extlinux/main.c:1131: undefined reference to 'minor'

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Gene Cumm <gene.cumm@gmail.com>
---
 extlinux/main.c | 1 +
 1 file changed, 1 insertion(+)

Index: syslinux-4.04/extlinux/main.c
===================================================================
--- syslinux-4.04.orig/extlinux/main.c
+++ syslinux-4.04/extlinux/main.c
@@ -37,6 +37,7 @@ typedef uint64_t u64;
 #include <sysexits.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/mount.h>
 #include <sys/vfs.h>
